corehq/apps/users/forms.py

```diff
@@ -209,24 +209,24 @@ def update_user(self, metadata_updated=False, profile_updated=False):
             self.existing_user.save()
 
         if is_update_successful and (props_updated or role_updated or metadata_updated):
-            messages = []
+            change_messages = {}
```

Comment:
There is actually [UserChangeLogger](https://github.com/dimagi/commcare-hq/blob/5fdfc6570d08acee9ec0a7ab19a8a6d43c377173/corehq/apps/user_importer/helpers.py#L21) that does that and should be used if we need to assemble multiple messages.
If we still need something to assemble messages, it would be better to add a new class for it, so that `UserChangeMessage` can just focus on making messages as its responsibility, though I think `UserChangeLogger` does that well by also adding messages [only when needed](https://github.com/dimagi/commcare-hq/blob/5fdfc6570d08acee9ec0a7ab19a8a6d43c377173/corehq/apps/user_importer/helpers.py#L53) and also [check on duplicates](https://github.com/dimagi/commcare-hq/blob/5fdfc6570d08acee9ec0a7ab19a8a6d43c377173/corehq/apps/user_importer/helpers.py#L71).

though it currently sits in `user_importer` app because that is where it was originally added but can definitely be moved to `users` app.